src/plugins/RepairMan.cpp

```diff
@@ -2,15 +2,57 @@
 //
 // Copyright (c) 2017-2020 Alexander Kurbatov
 
+#include "Historican.h"
+#include "Hub.h"
 #include "RepairMan.h"
 #include "core/API.h"
 #include "core/Helpers.h"
 
+namespace {
+Historican gHistory("repairman");
+
+}  // namespace
+
 void RepairMan::OnStep(Builder*) {
-    if (gAPI->observer().GetCurrentRace() != sc2::Race::Terran)
+    if (gHub->GetCurrentRace() != sc2::Race::Terran || m_damaged_buildings.empty())
+        return;
+
+    auto it = m_damaged_buildings.begin();
+    while (it != m_damaged_buildings.end()) {
+        if ((*it)->health == (*it)->health_max) {
+            gHistory.info() << sc2::UnitTypeToName((*it)->unit_type) <<
+            "(" << (*it)->tag << ") was repaired" << std::endl;
+            it = m_damaged_buildings.erase(it);
+            continue;
+        }
+
+        if (!(*it)->is_alive) {
+            m_damaged_buildings.erase(it);
+            continue;
+        }
+
+        ++it;
+    }
+
+    auto repairers = gAPI->observer().GetUnits(IsRepairer());
+    if (!m_damaged_buildings.empty() && repairers().empty()) {
+        gHub->AssignRepairTask(*m_damaged_buildings.front());
```

Comment:
It becomes idle and OnIdle will then call DistributeWorker to send it to nearest base to gather minerals. If the building is attacked again, it will still be close and will win the GetClosetFreeWorker and go to repair again.

For now, this avoids extra logic of staying near the building until the enemy leaves, but that can be done in the future for advanced Combat Repair micro.